src/components/BaseComponents/BaseHeader.tsx

```diff
@@ -44,13 +48,34 @@ export interface HeaderProps {
   rightIcon?: string;
   classes: any;
   match?: any;
+  name?: string;
+  email?: string;
 }
 
 function BaseHeader(props: HeaderProps) {
-  const { leftIcon, title, rightIcon, classes, match } = props;
+  const { leftIcon, title, rightIcon, classes, match, name, email } = props;
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
   const history = useHistory();
 
-  const getIcon = (onClick: (event: React.MouseEvent) => void, icon: JSX.Element) => {
+  const handleLogoutClick = async () => {
+    const logoutSuccess = await logoutUser();
+    if (logoutSuccess){
+      history.push('/login');
+    } else {
+      console.warn('Logout failed');
+    }
+  };
+
+  const openMenu = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  
+  const closeMenu = () => {
+    setAnchorEl(null);
+  };
```

Comment:
I'd specify `accountMenu` or something like that in case later menus get introduced.